docs/dapp/sapphire/precompiles.md

```diff
@@ -4,22 +4,61 @@ description: Additional Sapphire precompiles for encryption and confidentiality
 
 # Precompiles
 
-In addition to the standard EVM precompiles, Sapphire provides three
-further cryptography-related ones to make some operations easier and
-cheaper to perform: x25519 key derivation, and Deoxys-II-based encryption
-and decryption.
+In addition to the standard EVM precompiles, Sapphire provides a number
+of further cryptography-related ones to make some operations easier and
+cheaper to perform: x25519 key derivation, Deoxys-II-based encryption
+and decryption, signing key generation, message digest signing and
+verification.
 
 These can be called in the same way as other precompiles by dispatching
 calls to specific well-known contract addresses, as described below.
+
 Input parameters should be packed into a contiguous memory region, with
-each chunk of data padded to 32 bytes as usual.
+each chunk of data padded to 32 bytes as usual. Some precompiles have
+complex input parameter layouts, so it is recommended to use
+`abi.encode` and `abi.decode` to handle them; the signatures below are
+given with this in mind.
+
+## Library
+
+A few of the examples below show how to call the precompiles directly.
```

Comment:
I think we should just make all examples use the wrappers as otherwise people will just copy these.